modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/ReadOnlyTransactionImpl.java

```diff
@@ -39,17 +40,22 @@ class ReadOnlyTransactionImpl extends IgniteAbstractTransactionImpl {
     /** Prevents double finish of the transaction. */
     private final AtomicBoolean finishGuard = new AtomicBoolean();
 
+    /** The tracker is used to track an observable timestamp. */
+    private final HybridTimestampTracker observableTsTracker;
+
     /**
      * The constructor.
      *
      * @param txManager The tx manager.
+     * @param observableTsTracker Observable timestamp tracker.
      * @param id The id.
      * @param readTimestamp The read timestamp.
      */
-    ReadOnlyTransactionImpl(TxManagerImpl txManager, UUID id, HybridTimestamp readTimestamp) {
+    ReadOnlyTransactionImpl(TxManagerImpl txManager, HybridTimestampTracker observableTsTracker, UUID id, HybridTimestamp readTimestamp) {
         super(txManager, id);
 
         this.readTimestamp = readTimestamp;
+        this.observableTsTracker = observableTsTracker;
```

Comment:
Why we need it?
How do we update observableTsTracker in case of implicit RW transcion? I'd say that we should do it same way for implicit RO.